### PR TITLE
Add Repeat operator

### PR DESCRIPTION
**Pull request by ArekCzarnik**
#19 first proposal of RX "repeat create operation", we should go in discussion.

### CODE REVIEW

**Comment by jochasinga** [comment]
Thanks for your contribution. Could you check out the contributions page on our wiki and see if you can amend your commit summary/message to read better? For example, the summary should read only "Add Repeat operator".  Read the page for tips on how you can resolve issue using your commit message without explicitly including the issue # in the summary.

**Comment by ArekCzarnik** [comment]
Thanks, i read it and fix this today.

**Comment by ArekCzarnik** [comment]
@jochasinga 
current the link to contributions is broken, and the build have problem with missing tests in packages. Do you need help and should it try to fix it ?

missing link current:
https://github.com/ReactiveX/RxGo/blob/master/wiki/Contributions

File: observable/observable_test.go [hunk 1]
```diff
@@ -647,3 +647,99 @@ func TestObservableScanWithString(t *testing.T) {
 
 	assert.Exactly(t, expected, words)
 }
+
+func TestRepeateInfinityOperator(t *testing.T) {
```
Comment (jochasinga) [inline observable/observable_test.go]: There is a typo (extra 'e') in the func name.

File: observable/observable_test.go [hunk 2]
```diff
@@ -647,3 +647,99 @@ func TestObservableScanWithString(t *testing.T) {
...
+	}
+
+	if value, ok := item.(string); ok {
+		assert.Equal(t, value, "mystring")
+	} else {
+		assert.Fail(t, "fail to emit next item", err)
+	}
+}
+
+func TestRepeateNtimeOperator(t *testing.T) {
```
Comment (jochasinga) [inline observable/observable_test.go]: Here is a typo (extra 'e') after `Repeat` in the function name.

File: observable/observable_test.go [hunk 3]
```diff
@@ -647,3 +647,99 @@ func TestObservableScanWithString(t *testing.T) {
...
+		stringarray = append(stringarray, "end")
+	})
+
+	sub := myStream.Subscribe(observer.New(onNext, onDone))
+	<-sub
+
+	assert.Exactly(t, []string{"mystring", "mystring", "end"}, stringarray)
+}
+
+func TestRepeateNtimeMultiVariadicOperator(t *testing.T) {
```
Comment (jochasinga) [inline observable/observable_test.go]: Extra 'e' typo after `Repeat`

File: observable/observable_test.go [hunk 4]
```diff
@@ -647,3 +647,99 @@ func TestObservableScanWithString(t *testing.T) {
...
+		stringarray = append(stringarray, "end")
+	})
+
+	sub := myStream.Subscribe(observer.New(onNext, onDone))
+	<-sub
+
+	assert.Exactly(t, []string{"mystring", "mystring", "end"}, stringarray)
+}
+
+func TestRepeateWith0TimesOperator(t *testing.T) {
```
Comment (jochasinga) [inline observable/observable_test.go]: Extra 'e' typo and please change from using `0` in the function name to `Zero` instead.

File: observable/observable_test.go [hunk 5]
```diff
@@ -647,3 +647,99 @@ func TestObservableScanWithString(t *testing.T) {
...
+		stringarray = append(stringarray, "end")
+	})
+
+	sub := myStream.Subscribe(observer.New(onNext, onDone))
+	<-sub
+
+	assert.Exactly(t, []string{"end"}, stringarray)
+}
+
+func TestRepeateWithNegativeTimesOperator(t *testing.T) {
```
Comment (jochasinga) [inline observable/observable_test.go]: Extra 'e' typo

**Comment by ArekCzarnik** [comment]
Thanks for review the code, all typos should now be fix.

**Pull request merged**